src/server.rs

```diff
@@ -909,4 +912,97 @@ impl CryptoService for CryptoServiceServer {
         };
         Ok(tonic::Response::new(response))
     }
+
+    async fn secp256k1_priv_key_to_pub_key(
+        &self,
+        request: tonic::Request<Secp256k1PrivKeyToPubKeyRequest>,
+    ) -> std::result::Result<tonic::Response<Secp256k1PrivKeyToPubKeyResponse>, tonic::Status> {
+        let Secp256k1PrivKeyToPubKeyRequest { priv_key } = request.into_inner();
+        let priv_key: [u8; SECP256K1_PRIV_KEY_LENGTH] = priv_key
+            .ok_or_else(|| missing_field::<Secp256k1PrivKeyToPubKeyRequest>("priv_key"))?
```

Comment:
Nit: prefer `secret_key` to `private_key`. It is slightly stronger in meaning, and permits the convenient abbreviations `sk`/`pk` for secret key / public key